tendrl/gluster_integration/message/callback.py

```diff
@@ -9,14 +9,347 @@
+        native_event = NS.gluster.objects.NativeEvents(
+            context,
+            message=message,
+            severity="warning",
+            current_value="brick_path_resolve_failed",
+            alert_notify=True
+        )
+        native_event.save()
+
+    def quota_crossed_soft_limit(self, event):
+        context = "quota_crossed_soft_limit" + event[
+            "message"]["volume"] + event["message"]["path"]
+        message = "Quota soft limit crossed in volume: {0} for path: {1}" \
+                  ". Current usage: {2}".format(
+                      event['message']['volume'],
+                      event['message']['path'],
+                      event['message']['usage'],
+                  )
+        native_event = NS.gluster.objects.NativeEvents(
+            context,
+            message=message,
+            severity="warning",
+            current_value="quota_crossed_soft_limit",
+            alert_notify=True
+        )
+        native_event.save()
+
+    def bitrot_bad_file(self, event):
+        context = "bitrot_bad_file" + event['message'][
+            "brick"] + event['message']["path"] + event['message']["gfid"]
+        message = "File with gfid: {0} is corrupted due to bitrot." \
+                  "  Brick: {1}. Path: {2}".format(
+                      event['message']['gfid'],
+                      event['message']['brick'],
+                      event['message']['path'],
+                  )
+        native_event = NS.gluster.objects.NativeEvents(
+            context,
+            message=message,
+            severity="warning",
+            current_value="bitrot_bad_file",
+            alert_notify=True
+        )
+        native_event.save()
+
+    def afr_split_brain(self, event):
+        context = "afr_split_brain" + event['message']["subvol"]
+        message = "Subvolume: {} is affected by split-brain".format(
+            event['message']['subvol']
+        )
+        native_event = NS.gluster.objects.NativeEvents(
+            context,
+            message=message,
+            severity="warning",
+            current_value="afr_split_brain",
+            alert_notify=True
+        )
+        native_event.save()
+
+    def snapshot_soft_limit_reached(self, event):
+        context = "snapshot_soft_limit_reached" + event[
+            'message']['volume_name']
+        message = "Snapshot soft limit reached for volume: {}".format(
+            event['message']['volume_name']
+        )
+        native_event = NS.gluster.objects.NativeEvents(
+            context,
+            message=message,
+            severity="warning",
+            current_value="snapshot_soft_limit_reached",
+            alert_notify=True
+        )
+        native_event.save()
+
+    def snapshot_hard_limit_reached(self, event):
+        context = "snapshot_hard_limit_reached" + event[
+            'message']['volume_name']
+        message = "Snapshot hard limit reached for volume: {}".format(
```

Comment:
Does it return the limit values as well? If so we can add in message

Reply:
No